parser/ini.go

```diff
@@ -0,0 +1,120 @@
+package parser
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+)
+
+type Data map[string]map[string]string
+
+type Parser struct {
+	data Data
+}
+
+func NewParser() *Parser {
+	parser := Parser{}
+	parser.data = make(Data)
+	return &parser
+}
+
+func (parser *Parser) ReadFromString(content string) {
+	lines := strings.Split(content, "\n")
+	var section string
+
+	for _, line := range lines {
+		line = strings.TrimSpace(line)
+		if len(line) > 2 && line[0] == '[' && line[len(line)-1] == ']' {
+
+			section = strings.TrimSpace(line[1 : len(line)-1])
+			parser.data[section] = map[string]string{}
+
+		} else if strings.Contains(line, "=") {
+			keyVal := strings.Split(line, " = ")
+			key := keyVal[0]
+			val := keyVal[1]
+			parser.data[section][key] = val
+
+		}
+	}
+}
+
+func (parser *Parser) ReadFile(path string) error {
```

Comment:
we can abstract the core logic in a function `readFromReader` then use it in ReadFromString and ReadFromFile functions